src/SequenceManager.cc

```diff
@@ -94,30 +96,36 @@ void store_sequence_parameters(const std::filesystem::path& lua_file, const Sequ
 } // anonymous namespace
 
 SequenceManager::SequenceManager(std::filesystem::path path)
-    : path_{ std::move(path) }
+    : path_{ std::move(path) }, gl_{path_}
 {
     if (path_.empty())
         throw Error("Base path name for sequences must not be empty");
 }
 
 Sequence
-SequenceManager::copy_sequence(UniqueId original_uid, const SequenceName& new_name) const
+SequenceManager::copy_sequence(UniqueId original_uid, const SequenceName& new_name)
 {
     const auto sequences = list_sequences();
     const UniqueId new_unique_id = create_unique_id(sequences);
 
     Sequence sequence = load_sequence(original_uid, sequences);
+    const std::string& old_name = sequence.get_name().string();
 
     sequence.set_unique_id(new_unique_id);
     sequence.set_name(new_name);
 
-    store_sequence(sequence);
+    _store_sequence(sequence);
+
+    // commit to local repository
+    const auto new_folder_name = make_sequence_filename(new_name, new_unique_id);
+    gl_.add_files({gul14::cat(new_folder_name, "/*")});
+    gl_.commit(gul14::cat("Copy ", old_name, " to ", new_name.string()));
```

Comment:
Just an idea: Instead of calling `_store_sequence(sequence)` and afterwards committing with a "special" commit message, maybe we could have a private function `store_sequence_with_commit_message(sequence, commit_msg)`? This way we would not have to reconstruct the sequence filename here.

Reply:
outdated code